File: 실버/11723(집합)/Main.cpp
#include <iostream>
#include <cstring> // strcmp
#include <string>	// string type
#include <map>
using namespace std;

int main() {
	// IO 속도 향상
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	
	int M=0, num=0, check=0;
	long set = 0;
	char cmd[7];
	map<string, int> m;
	m["add"] = 0; m["remove"] = 1; m["check"] = 2;
	m["toggle"] = 3; m["all"] = 4; m["empty"] = 5;

	cin >> M;
	for (int i = 0; i < M; i++) {
		cin >> cmd;
		if (strcmp(cmd, "all") && strcmp(cmd, "empty")) {
			cin >> num;
		}
		check = 1 << (num - 1);
		switch (m[cmd]) {
			case 0:
				set |= check;
				break;
			case 1:
				set &= ~check;
				break;
			case 2:
				if ((set & check) == check)
					cout << 1 << "\n";
				else
					cout << 0 << "\n";
				break;
			case 3:
				if ((set & check) == check)
					set &= ~check;
				else
					set |= check;
				break;
			case 4:
				set = (1 << 20) - 1;
				break;
			case 5:
				set = 0;
				break;
		}
	}
	return 0;
}

File: 실버/11723(집합)/Main.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

class Main {
	public static void main(String[] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		//StringBuilder sb = new StringBuilder();
		BufferedWriter bw =  new BufferedWriter(new OutputStreamWriter(System.out));
		int M = Integer.parseInt(br.readLine());
		long set = 0;

		int num = 0;
		int check = 0;
		for(int i=0; i<M; i++){
			String[] temp = br.readLine().split(" ");
			if(!temp[0].equals("empty") && !temp[0].equals("all"))
				num = Integer.parseInt(temp[1]);
			check = 1 << (num-1);
			switch(temp[0]){
				case "add":
					set |= check;
					break;
				case "remove":
					set &= ~check;
					break;
				case "check":
					if((set & check) == check)
						bw.write(1 + "\n");
					else
						bw.write(0 + "\n");
					break;
				case "toggle":
					if((set & check) == check)
						set &= ~check;
					else
						set |= check;
					break;
				case "all":
					set = (1 << 20) - 1;
					break;
				case "empty":
					set = 0;
					break;
			}
		}
		bw.flush();
		bw.close();
		br.close();
	}

}